compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/IrInlineCodegen.kt

```diff
@@ -152,23 +147,14 @@ class IrExpressionLambdaImpl(
     }
 
     companion object {
-        private var counter: Int = 123//TODO: pass proper type
+        private var counter: Int = 1//TODO: pass proper type
     }
 
-    override val lambdaClassType: Type = Type.getObjectType("test${counter++}")
+    override val lambdaClassType: Type = Type.getObjectType("$typePrefix\$lambda-${counter++}")
```

Comment:
Maybe best to use precalculated old name ```codegen.context.getLocalClassInfo(reference)```

Reply:
There still needs to be a fallback in case the reference was generated by a lowering after InventNamesForLocalClasses (e.g. by InlineCallableReferenceToLambdaPhase) and has no LocalClassInfo. Since this name never appears in the output and any unique string is fine, IMO it's not worth the effort.

Reply:
Actually seems it's only InlineCallableReferenceToLambdaPhase lower that was writen especcially for inline purpose. It could be simply patched in 2 places via:
```
+IrFunctionReferenceImpl(...).also {
                        this@InlineCallableReferenceToLambdaPhase.context.putLocalClassInfo(
                            it,
                            this@InlineCallableReferenceToLambdaPhase.context.getLocalClassInfo(expression)!!
                        )
                    }
```
It's worth for debugging purpose in case of two backend support

Reply:
Seems best to use: ```it.copyAttributes(expression)```

Reply:
Done